components-core/src/main/java/com/adyen/checkout/components/base/BaseConfigurationBuilder.kt

```diff
@@ -0,0 +1,57 @@
+package com.adyen.checkout.components.base
+
+import android.content.Context
+import com.adyen.checkout.components.util.ValidationUtils.isClientKeyValid
+import com.adyen.checkout.core.api.Environment
+import com.adyen.checkout.core.exception.CheckoutException
+import com.adyen.checkout.core.util.LocaleUtil
+import java.util.*
+
+/**
+ * Base constructor with the required fields.
+ *
+ * @param shopperLocale The Locale of the shopper.
+ * @param environment   The {@link Environment} to be used for network calls to Adyen.
+ * @param clientKey Your Client Key used for network calls from the SDK to Adyen.
+ */
+abstract class BaseConfigurationBuilder<ConfigurationT : Configuration>(
+    var builderShopperLocale: Locale,
```

Comment:
These fields should be `protected` I think.

Reply:
👍

Reply:
@caiofaustino's point in one of the previous comments made sense to me since it was already possible to modify them via setters I don't think it makes too much of a difference. @jreij why do you think they should be protected?

Reply:
oh yeah right, in the builder they could change the values whenever, even if they already passed them in the constructor it's fine to let them change after, so you I guess public is better indeed

Reply:
It's true that they can change the values but they couldn't read them (we didn't have getters). And besides, the methods we have are not actual setters, they are builder methods (they return the Builder), so i feel like it's confusing for someone to see that they can call `Builder.serShopperLocale` and also `Builder.builderShopperLocale = ...`.

It's just a pattern thing though, not a big deal.

Reply:
You are totally right Joseph.

Reply:
got it 👍